### PR TITLE
Fix crash to render failed message warning stub

### LINKED ISSUE #4110: Crash while rendering failed message warning
### Steps to reproduce

- Enable flight mode
- Send a message (easy to repro with media messages)
- Restart app
- Open the room which has failed messages

### What did you expect?
To see the timeline with failed messages warning

### What happened?
App crashed.

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

_No response_

### Homeserver

_No response_

### Have you submitted a rageshake?

No

### PR DESCRIPTION
**Pull request by onurays**
Fixes #4110

### CODE REVIEW

File: changelog.d/3833.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+[Bug|Crash] Voice Messages sending need more robustfully
```
Comment (bmarty) [inline changelog.d/3833.bugfix:1]: This is not really related to voice message, and/or issue 3833.

**Comment by ganfra** [comment]
> Fixes #3833

I don't see how it's fixing 3833?
Your PR is fixing a crash introduced after the issue was created

File: vector/src/main/res/layout/fragment_room_detail.xml [hunk 2]
```diff
@@ -98,7 +98,7 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         app:barrierDirection="top"
-        app:constraint_referenced_ids="composerLayout,notificationAreaView,failedMessagesWarningView" />
+        app:constraint_referenced_ids="composerLayout,notificationAreaView,failedMessagesWarningStub" />
```
Comment (ganfra) [inline vector/src/main/res/layout/fragment_room_detail.xml:101]: I think the other Barrier needs to update too

**Review by ganfra — CHANGES_REQUESTED** [review]
See my remarks

**Review by bmarty — CHANGES_REQUESTED** [review]
Thanks for the fix, crash reproduced and fix tested OK, but the changelog file is not correct.

Can you update it please?
It should be file `4105.bugfix` with content "Fix a crash (regression in 1.3.0) in the timeline when a message has failed to be sent"

File: vector/src/main/res/layout/fragment_room_detail.xml (hunk 2 above)
Comment (bmarty) [reply vector/src/main/res/layout/fragment_room_detail.xml:101]: It's already `failedMessagesWarningStub`, do you confirm it's wrong?
https://github.com/vector-im/element-android/blob/develop/vector/src/main/res/layout/fragment_room_detail.xml#L207

Comment (onurays) [reply vector/src/main/res/layout/fragment_room_detail.xml:101]: Yes, it is.

Comment (bmarty) [reply vector/src/main/res/layout/fragment_room_detail.xml:101]: it is... wrong?

**Review by bmarty — APPROVED** [review]

**Pull request merged**